src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/VirtualFileResultExecutor.cs

```diff
@@ -110,6 +110,11 @@ protected virtual Task WriteFileAsync(ActionContext context, VirtualFileResult r
         private IFileInfo GetFileInformation(VirtualFileResult result)
         {
             var fileProvider = GetFileProvider(result);
+            if (fileProvider is NullFileProvider)
```

Comment:
I ran into this while trying to download a file. All I got was file not found (which itself is from this NullFileProvider).

Reply:
We have precedence for doing something like this - https://github.com/aspnet/Mvc/blob/452578e4a880676a35a8c50a6e4e8e11dccbd125/src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewCompilerProvider.cs#L50. Seems ok to me

Reply:
Great 👍